packages/blocks/src/store/process-block-type.js

```diff
@@ -197,5 +197,41 @@ export const processBlockType =
 			return;
 		}
 
+		if (
+			typeof settings?.parent === 'string' ||
```

Comment:
Copied the comment from the other PR:

> > I'm curious why we'd need both typeof blockType.parent === 'string' || blockType.parent instanceof String.. 

> Otherwise, we don't cover all potential issues. Note the issue was that we checked for `.length` in a type that wasn't an array (didn't have `.some`). That's the case for both literal & object strings.

I still don't see why is needed. Can you give me an example?

In general my [suggestion before ](https://github.com/WordPress/gutenberg/pull/66250#discussion_r1810049802)for simplifying the checks by what we expect and not what we don't expect. With that code `parent` could be a boolean etc.. 

And if we wanted to be thorough we should also check if an array is provided, that consists of `strings`.

Reply:
Ok, these are the facts:

- I discovered that we [were trying to detect arrays the wrong way](https://github.com/WordPress/gutenberg/pull/66234/files#diff-aa92c48095c35f17feb834dd919baf52274a73c7b1180ff5acce7b6087b126cfL1631) while responding to [Gutenberg forum issues](https://wordpress.org/support/topic/error-after-update-228/). We were checking for `blockType.parent?.length`, but that's not enough to detect arrays — it can be a string, any type of it, as it was. I prepared a [hotfix](https://github.com/WordPress/gutenberg/pull/66234) but wanted to tackle the core issue: that we didn't normalize the parent data coming from 3rd parties — that's what this PR is for.  I presume we have more of these in our codebase are causing issues for the same reason.

- Strings can be [literal or object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#string_primitives_and_string_objects). Both have the `.length` property and are [boxed/unboxed](https://oandre.gal/articles/values-and-references-in-javascript/#values-references) depending on what you do. So 

```js
console.log( 'string'.length ); // Logs 6
console.log( new String( 'string' ).length ); // Logs 6
```

log the same. Block authors should have been using arrays to declare the parent, but, in some cases, they weren't — instead they used strings of any type. When registering a block you can use either, that's why we need to cover for both.

Does this help?